project-management/src/main/java/life/qbic/projectmanagement/application/sample/SampleRegistrationServiceV2.java

```diff
@@ -42,32 +51,46 @@ public class SampleRegistrationServiceV2 {
   private final BatchRepository batchRepository;
   private final SampleCodeService sampleCodeService;
   private final DeletionService deletionService;
+  private final ConfoundingVariableService confoundingVariableService;
 
   @Autowired
   public SampleRegistrationServiceV2(BatchRegistrationService batchRegistrationService,
       SampleRepository sampleRepository, BatchRepository batchRepository,
       SampleCodeService sampleCodeService,
-      DeletionService deletionService) {
+      DeletionService deletionService, ConfoundingVariableService confoundingVariableService) {
     this.batchRegistrationService = Objects.requireNonNull(batchRegistrationService);
     this.sampleRepository = Objects.requireNonNull(sampleRepository);
     this.batchRepository = Objects.requireNonNull(batchRepository);
     this.sampleCodeService = Objects.requireNonNull(sampleCodeService);
     this.deletionService = Objects.requireNonNull(deletionService);
+    this.confoundingVariableService = confoundingVariableService;
   }
 
   @PreAuthorize("hasPermission(#projectId, 'life.qbic.projectmanagement.domain.model.project.Project', 'WRITE')")
   @Async
+  @Transactional(propagation = Propagation.REQUIRES_NEW)
   public CompletableFuture<Void> registerSamples(Collection<SampleMetadata> sampleMetadata,
-      ProjectId projectId, String batchLabel, boolean batchIsPilot, ExperimentId experimentId)
+      ProjectId projectId, String batchLabel, boolean batchIsPilot, ExperimentReference experiment)
       throws RegistrationException {
     var result = batchRegistrationService.registerBatch(batchLabel, batchIsPilot, projectId,
-        experimentId);
+        ExperimentId.parse(experiment.id()));
```

Comment:
I think we overdid it here big time with the Identifier... Imo there is no need to parse, because it does not have any semantics or schema. It is just a string value generated from the UUID generator. You either find an object with the ID in the database, or you dont. 
Happy to see some simplification.

Reply:
I agree!